coding styles (#151)
r-hub/blog: content/post/2022-02-21-code-style/index.md

## Etiquette

If you are a contributor to a codebase, you'll probably have to adapt your style to the maintainer's or maintainers' preferences. These preferences might be implicit (seen by reading existing code) or explicit (in a contributing guide). Now, depending on your relation to the maintainers (e.g. is it your first contribution), you might start a discussion about changing some of the conventions, perhaps changing their mind or reaching a compromise (assigning with `=`, but with spaces before and after each equal sign).

If you are the maintainer of a codebase, you'll need to be forgiving when stating your preferences and explaining how to implement them; or you might even want to take matters in your own hands and do some [restyling yourself](https://yihui.org/en/2018/11/cosmetic-changes/).

## Conclusion

In this post we shared documentation of and tooling for coding style. We hope it can help you write or advocate for more readable code. Practice makes perfect, so go forth and participate in code production and reviews. :wink:

> [^1]: Note that the Google R style guide inspired the tidyverse styleguide but Google now refers to the tidyverse style guide as R style guide.

Reviewer: ```suggestion
[^1]: Note that the Google R style guide first inspired the tidyverse styleguide but now Google R style guide is derived from the tidyverse style guide. 
```
